design/spec/upsert/protocol.md

```
@@ -0,0 +1,166 @@
+|------|-------------------|----------------|
+| Explicit leaf (keyed) | `leaf->data + key_size` | Leaf's `optimistic_lock` |
+| Value-in-slot (VIS/packed) | Child pointer in inode | Containing inode's `optimistic_lock` |
+| Root leaf | Root pointer | `root_pointer_lock` |
+
+[source: §11.1]
+
+---
+
+## Keyed-Leaf Update Protocol
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)